src/lib/optimizer/strategy/or_to_union_rule.cpp

```diff
@@ -0,0 +1,34 @@
+#include <expression/expression_utils.hpp>
+
+#include "expression/logical_expression.hpp"
+#include "logical_query_plan/lqp_utils.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "or_to_union_rule.hpp"
+
+namespace opossum {
+
+std::string OrToUnionRule::name() const { return "Or to Union Rule"; }
+
+void OrToUnionRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) const {
+  if (node->type == LQPNodeType::Predicate) {
+    const auto predicate_node = std::static_pointer_cast<PredicateNode>(node);
+    const auto predicate = predicate_node->predicate();
+
+    const auto flat_disjunction = flatten_logical_expressions(predicate, LogicalOperator::Or);
+    if (flat_disjunction.size() == 2) {
```

Comment:
you shouldn't have to limit this to 2, disjunctive chains of any size >= 2 can be reformulated into a series of unions